### PR TITLE
Adds Kryo serializer for ProtocolBuffer classes

### PR DESCRIPTION
**Pull request by Crim**

### CODE REVIEW

File: src/main/java/de/javakaffee/kryoserializers/protobuf/ProtobufSerializer.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        // If the length is equal to our special null value
+        if (length == NULL) {
+            // Just return null
+            return null;
+        }
+        // Otherwise read the byte array length
+        byte[] bytes = input.readBytes(length - 1);
+        try {
+            // Deserialize protobuf
+            Method method = type.getMethod("parseFrom", byte[].class);
```
Comment (hepin1989) [inline src/main/java/de/javakaffee/kryoserializers/protobuf/ProtobufSerializer.java]: need cache the method look up.

Comment (Crim) [inline src/main/java/de/javakaffee/kryoserializers/protobuf/ProtobufSerializer.java]: Updated to cache the reflection call, and expanded the tests a bit to verify it works as expected across multiple protobuf instances/types.

**Comment by Crim** [comment]
Ok, all good I think :)

**Comment by hepin1989** [comment]
you could take a look at 
https://github.com/akka/akka/blob/master/akka-remote/src/main/scala/akka/remote/serialization/ProtobufSerializer.scala
:)

**Comment by Crim** [comment]
I'll have to admit I'm not the best with reading Scala, that implementation looks similar to what I did?  Or am I missing something?

**Comment by magro** [comment]
They cache the parseFrom method per class, that's the main difference I'd say. Not sure if this is needed, I don't know how protobuf works.

**Comment by Crim** [comment]
I can add a static map linking class name => cached reflection method.  As it stands now within an instance of the serializer it should be caching the call, but if you had two separate instances of the same serializer for the same protobuf class, you'd have a slight performance hit.

**Comment by magro** [comment]
If the map would be static it would have to be a thread safe / concurrent map.
What I'm just realizing: because kryo holds a serializer instance per class, you could leave it as it is, with just the cached method. Right?

**Comment by Crim** [comment]
That was exactly my thought -^

**Comment by magro** [comment]
Ahh, ok :-)

**Comment by magro** [comment]
I've one question left: How are the protobuf files compiled to classes? I didn't see any instructions to generate them, what I'd have expected...

**Comment by hepin1989** [comment]
@magro you should call the ```protoc``` to compile it.
and there is an annotation based one,which map the java class to protobuf style one.
take a look at this
https://github.com/jhunters/jprotobuf

I think for this issue,we could just using the AtomicReference<Map<Class,Method>> or the ConcurrentHashMap<Class,Method> and just return the ```Message```.

**Comment by Crim** [comment]
So instructions for compiling the .proto schema files into Java classes is explained on the protobufs github Readme found here: https://github.com/google/protobuf/tree/master/java

**Comment by magro** [comment]
Ok, thanks. @Crim can you stash the commits? Then I'll merge and release.

**Comment by Crim** [comment]
squashed the commits

**Comment by Crim** [comment]
*bump* :D

**Comment by magro** [comment]
Sorry, I was a bit busy the last days, now it's released as 0.30 (on it's way to maven central). And thanks for the PR!! :-)

**Pull request merged**